src/components/generics/vacation-test-forms/employee-vacation-requests.tsx

```diff
@@ -360,43 +397,102 @@ const RenderEmployeeVacationRequests = ({ persons }: Props) => {
   // };
 
   /**
-   * Handle request type
-   *
-   * @param type Vacation type
+   * Update vacation request status
+   * 
+   * @param selectedStatusObject id of the vacation request status
+   * @param newStatus status to be applied to the updated vacation request status
    */
-  const handleRequestType = (type: VacationType) => {
-    switch (type) {
-      case VacationType.VACATION:
-        return strings.vacationRequests.vacation;
-      case VacationType.PERSONAL_DAYS:
-        return strings.vacationRequests.personalDays;
-      case VacationType.UNPAID_TIME_OFF:
-        return strings.vacationRequests.unpaidTimeOff;
-      case VacationType.MATERNITY_PATERNITY:
-        return strings.vacationRequests.maternityPaternityLeave;
-      case VacationType.SICKNESS:
-        return strings.vacationRequests.sickness;
-      case VacationType.CHILD_SICKNESS:
-        return strings.vacationRequests.childSickness;
-      default:
-        return strings.vacationRequests.vacation;
+  const updateVacationRequestStatus = async (selectedStatusObject: VacationRequestStatus, newStatus: VacationRequestStatuses) => {
+    if (!selectedStatusObject) {
+      return;
+    }
+
+    try {
+      const updateApi = Api.getVacationRequestStatusApi(accessToken?.access_token);
+      const updatedRequestStatus = await updateApi.updateVacationRequestStatus({
+        id: selectedStatusObject.id!,
+        statusId: selectedStatusObject.id!,
+        vacationRequestStatus: {
+          ...selectedStatusObject,
+          status: newStatus,
+          vacationRequestId: selectedStatusObject.vacationRequestId,
+          createdAt: selectedStatusObject.createdAt,
+          createdBy: selectedStatusObject.createdBy,
+          message: selectedStatusObject.message,
+          updatedAt: new Date(),
+          updatedBy: person?.keycloakId
+        }
+      });
+
+      const update = latestRequestStatuses.map(requestStatus => (requestStatus.id !== selectedStatusObject.id ? requestStatus : updatedRequestStatus));
+      setLatestStatuses(update);
+    } catch (error) {
+      context.setError(strings.errorHandling.fetchVacationDataFailed, error);
     }
   };
 
-  // /**
-  //  * Handle request status
-  //  *
-  //  * @param requestStatus Vacation request status
-  //  */
-  // const handleRequestStatus = (requestStatus: VacationRequestStatuses) => {
-  //   const statusMap = {
-  //     [VacationRequestStatuses.PENDING]: strings.vacationRequests.pending,
-  //     [VacationRequestStatuses.APPROVED]: strings.vacationRequests.approved,
-  //     [VacationRequestStatuses.DECLINED]: strings.vacationRequests.declined
-  //   };
-
-  //   return statusMap[requestStatus] || "";
-  // };
+  /**
+   * Handle request status on click
+   * 
+   * @param approved is status approved or not
+   * @param requestId the id of vacation request
+   */
+  const handleClick = (approved: boolean, requestId: string | undefined) => {
```

Comment:
More descriptive function name

Reply:
Changed function name from "handleStatus" to "changeVacationRequestStatus"
```suggestion
    const selectedStatusIndex = latestRequestStatuses.findIndex(s => s.vacationRequestId === requestId);
```